src/components/Pressable.tsx

```diff
@@ -0,0 +1,281 @@
+
+  /**
+   * Either children or a render prop that receives a boolean reflecting whether
+   * the component is currently pressed.
+   */
+  children?:
+    | React.ReactNode
+    | ((state: PressableStateCallbackType) => React.ReactNode);
+
+  /**
+   * Whether a press gesture can be interrupted by a parent gesture such as a
+   * scroll event. Defaults to true.
+   */
+  cancelable?: null | boolean;
+
+  /**
+   * Duration to wait after hover in before calling `onHoverIn`.
+   * @platform macos windows
+   */
+  delayHoverIn?: number | null;
+
+  /**
+   * Duration to wait after hover out before calling `onHoverOut`.
+   * @platform macos windows
+   */
+  delayHoverOut?: number | null;
+
+  /**
+   * Duration (in milliseconds) from `onPressIn` before `onLongPress` is called.
+   */
+  delayLongPress?: null | number;
+
+  /**
+   * Whether the press behavior is disabled.
+   */
+  disabled?: null | boolean;
+
+  /**
+   * Additional distance outside of this view in which a press is detected.
+   */
+  hitSlop?: null | Insets | number;
+
+  /**
+   * Additional distance outside of this view in which a touch is considered a
+   * press before `onPressOut` is triggered.
+   */
+  pressRetentionOffset?: null | Insets | number;
+
+  /**
+   * If true, doesn't play system sound on touch.
+   */
+  android_disableSound?: null | boolean;
+
+  /**
+   * Enables the Android ripple effect and configures its color.
+   */
+  android_ripple?: null | PressableAndroidRippleConfig;
+
+  /**
+   * Used only for documentation or testing (e.g. snapshot testing).
+   */
+  testOnly_pressed?: null | boolean;
+
+  /**
+   * Either view styles or a function that receives a boolean reflecting whether
+   * the component is currently pressed and returns view styles.
+   */
+  style?:
+    | StyleProp<ViewStyle>
+    | ((state: PressableStateCallbackType) => StyleProp<ViewStyle>);
+
+  /**
+   * Duration (in milliseconds) to wait after press down before calling onPressIn.
+   */
+  unstable_pressDelay?: number;
+}
+
+function touchWithinBounds(touch: TouchData, bounds: Insets): boolean {
+  const leftbound = bounds.left ? bounds.left < touch.absoluteX : true;
+  const rightbound = bounds.right ? bounds.right > touch.absoluteX : true;
+  const bottombound = bounds.bottom ? bounds.bottom < touch.absoluteY : true;
+  const topbound = bounds.top ? bounds.top > touch.absoluteY : true;
+
+  return leftbound && rightbound && topbound && bottombound;
+}
+
+export default function Pressable(props: PressableProps) {
+  const previousTouchData = useRef<TouchData[] | null>(null);
+
+  const pressRetentionOffset: Insets | null | undefined =
+    typeof props.pressRetentionOffset === 'number'
+      ? {
+          top: props.pressRetentionOffset,
+          left: props.pressRetentionOffset,
+          bottom: props.pressRetentionOffset,
+          right: props.pressRetentionOffset,
+        }
+      : props.pressRetentionOffset;
+
+  const touch = Gesture.Manual()
```

Comment:
It's very possible, that we should be using `Gesture.Native` here combined with `GestureHandlerButton` for tracking touches (I think there may be differences when it comes to handling hitslop), though that needs to be checked